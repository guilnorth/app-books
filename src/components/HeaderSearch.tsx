import React, { useContext } from 'react';
import { Keyboard } from 'react-native';
import { Button, Div, Icon, Input, Text } from 'react-native-magnus';
import COLORS from '../styles/colors';
import Metrics from '../styles/metrics';
import { BooksContext } from '../context/BookContext';

export default function HeaderSearch() {
  const { searchBooks, term, setTerm } = useContext(BooksContext);
  const handleSearch = (keepList = false) => {
    searchBooks(keepList);
    Keyboard.dismiss();
  };
  return (
    <Div mb={20} alignSelf="flex-start" mx={Metrics.mediumSpacing} mt={20}>
      <Text
        fontWeight="bold"
        fontSize={28}
        textAlign="left"
        color="white"
        lineHeight={32}
        mb={4}
      >
        Busca
      </Text>
      <Div
        style={{
          borderBottomColor: COLORS.white,
          borderBottomWidth: 1,
          width: 40,
        }}
      />

      <Input
        testID="inputSearch"
        mt={40}
        bg={COLORS.bgPage}
        style={{ width: '100%' }}
        placeholder="Pesquise pelo livro aqui"
        borderColor={COLORS.white}
        fontSize={16}
        color="white"
        focusBorderColor={COLORS.white}
        returnKeyType="search"
        suffix={
          <Button
            testID="submitSearch"
            bg="transparent"
            h={40}
            w={40}
            rounded="circle"
            onPress={() => {
              handleSearch();
            }}
          >
            <Icon name="search" color="white" fontFamily="Feather" />
          </Button>
        }
        onSubmitEditing={() => {
          handleSearch();
        }}
        value={term}
        onChangeText={(text) => {
          setTerm(text);
        }}
        clearButtonMode="while-editing"
      />
    </Div>
  );
}
