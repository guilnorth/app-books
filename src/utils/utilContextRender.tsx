import React from 'react';
import { render } from '@testing-library/react-native';
import { BooksProvider, BooksContext } from '../context/BookContext';

const renderWithContext = (component: any) => ({
  ...render(<BooksProvider value={BooksContext}>{component}</BooksProvider>),
});

export { BooksProvider, renderWithContext };
