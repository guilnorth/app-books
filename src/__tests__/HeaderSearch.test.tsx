import * as React from 'react';
import { cleanup, fireEvent, act, render } from '@testing-library/react-native';
import { renderWithContext } from '../utils/utilContextRender';
import HeaderSearch from '../components/HeaderSearch';
import App from '../../App';

afterEach(() => {
  cleanup();
  jest.clearAllMocks();
});

const mockHooks = jest.spyOn(React, 'useContext');
mockHooks.mockImplementationOnce(() => ({
  term: '',
  setTerm: () => {},
  searchBooks: () => {},
}));

test('Button Submit to be defined', () => {
  const { getByTestId } = renderWithContext(<HeaderSearch />);
  const btnSubmit = getByTestId('submitSearch');
  expect(btnSubmit).toBeDefined();
});

test('Search input to be defined', () => {
  const { getByTestId } = renderWithContext(<HeaderSearch />);
  const inputSearch = getByTestId('inputSearch');
  expect(inputSearch).toBeDefined();
});

test('Call to be 1', () => {
  const { getByTestId } = renderWithContext(<HeaderSearch />);
  const inputSearch = getByTestId('inputSearch');
  const btnSubmit = getByTestId('submitSearch');
  act(() => {
    fireEvent.changeText(inputSearch, 'Harry');
    fireEvent.press(btnSubmit);
    expect(mockHooks.mock.calls.length).toBe(1);
  });
});

describe('HeaderSearch renders correctly', () => {
  it('App', async () => {
    const component = render(<HeaderSearch />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
