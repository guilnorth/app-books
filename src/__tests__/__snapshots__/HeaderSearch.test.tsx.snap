// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderSearch renders correctly App 1`] = `
<View
  alignSelf="flex-start"
  borderStyle="solid"
  mx={16}
  pointerEvents="auto"
  style={
    Object {
      "alignItems": undefined,
      "alignSelf": "flex-start",
      "backgroundColor": "rgba(0,0,0,0)",
      "borderRadius": 0,
      "borderStyle": "solid",
      "flex": undefined,
      "flexDirection": "column",
      "flexWrap": "nowrap",
      "height": undefined,
      "justifyContent": undefined,
      "marginBottom": 20,
      "marginHorizontal": 16,
      "marginTop": 20,
      "maxHeight": undefined,
      "maxWidth": undefined,
      "minHeight": undefined,
      "minWidth": undefined,
      "opacity": undefined,
      "overflow": undefined,
      "position": "relative",
      "shadowColor": undefined,
      "width": undefined,
      "zIndex": undefined,
    }
  }
>
  <Text
    style={
      Object {
        "backgroundColor": undefined,
        "color": "#FFFFFF",
        "flex": undefined,
        "fontFamily": undefined,
        "fontSize": 28,
        "fontStyle": undefined,
        "fontWeight": "bold",
        "height": undefined,
        "letterSpacing": undefined,
        "lineHeight": 32,
        "marginBottom": 4,
        "maxHeight": undefined,
        "maxWidth": undefined,
        "minHeight": undefined,
        "minWidth": undefined,
        "opacity": undefined,
        "overflow": "hidden",
        "textAlign": "left",
        "textAlignVertical": "center",
        "textDecorationColor": undefined,
        "textDecorationLine": undefined,
        "textDecorationStyle": undefined,
        "textShadowColor": undefined,
        "textShadowOffset": Object {
          "height": undefined,
          "width": undefined,
        },
        "textShadowRadius": undefined,
        "textTransform": "none",
        "width": undefined,
      }
    }
  >
    Busca
  </Text>
  <View
    borderStyle="solid"
    pointerEvents="auto"
    style={
      Object {
        "alignItems": undefined,
        "alignSelf": undefined,
        "backgroundColor": "rgba(0,0,0,0)",
        "borderBottomColor": "#FFFFFF",
        "borderBottomWidth": 1,
        "borderRadius": 0,
        "borderStyle": "solid",
        "flex": undefined,
        "flexDirection": "column",
        "flexWrap": "nowrap",
        "height": undefined,
        "justifyContent": undefined,
        "maxHeight": undefined,
        "maxWidth": undefined,
        "minHeight": undefined,
        "minWidth": undefined,
        "opacity": undefined,
        "overflow": undefined,
        "position": "relative",
        "shadowColor": undefined,
        "width": 40,
        "zIndex": undefined,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#2B2B2B",
        "borderColor": "#FFFFFF",
        "borderRadius": 6,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": undefined,
        "marginTop": 40,
        "maxHeight": undefined,
        "maxWidth": undefined,
        "minHeight": undefined,
        "minWidth": undefined,
        "opacity": undefined,
        "paddingHorizontal": 12,
        "paddingVertical": 12,
        "width": "100%",
        "zIndex": undefined,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      clearButtonMode="while-editing"
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      onSubmitEditing={[Function]}
      placeholder="Pesquise pelo livro aqui"
      placeholderTextColor="rgba(255, 255, 255, 0.4)"
      px="lg"
      py="lg"
      rejectResponderTermination={true}
      returnKeyType="search"
      style={
        Object {
          "color": "#FFFFFF",
          "flex": 1,
          "fontFamily": undefined,
          "fontSize": 16,
          "fontStyle": undefined,
          "fontWeight": undefined,
          "letterSpacing": undefined,
          "lineHeight": undefined,
          "padding": 0,
          "textAlign": undefined,
          "textAlignVertical": undefined,
          "textDecorationColor": undefined,
          "textDecorationLine": undefined,
          "textDecorationStyle": undefined,
          "textShadowColor": undefined,
          "textShadowOffset": Object {
            "height": undefined,
            "width": undefined,
          },
          "textShadowRadius": undefined,
          "textTransform": undefined,
        }
      }
      testID="inputSearch"
      underlineColorAndroid="transparent"
    />
    <View
      style={
        Object {
          "alignSelf": undefined,
          "flexDirection": "row",
          "justifyContent": "center",
          "marginLeft": 5,
        }
      }
    >
      <View
        accessible={true}
        alignItems="center"
        flexDir="row"
        focusable={true}
        justifyContent="center"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        position="relative"
        style={
          Array [
            Object {
              "alignItems": "center",
              "alignSelf": "flex-start",
              "backgroundColor": "rgba(0,0,0,0)",
              "borderRadius": 99999,
              "flex": undefined,
              "flexDirection": "row",
              "height": 40,
              "justifyContent": "center",
              "maxHeight": undefined,
              "maxWidth": undefined,
              "minHeight": undefined,
              "minWidth": undefined,
              "opacity": undefined,
              "overflow": "hidden",
              "padding": 12,
              "position": "relative",
              "width": 40,
              "zIndex": undefined,
            },
            false,
          ]
        }
        testID="submitSearch"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
              "position": "relative",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": undefined,
                "backgroundColor": undefined,
                "bottom": undefined,
                "color": "#FFFFFF",
                "fontSize": 13,
                "height": undefined,
                "justifyContent": "center",
                "left": undefined,
                "maxHeight": undefined,
                "maxWidth": undefined,
                "minHeight": undefined,
                "minWidth": undefined,
                "opacity": undefined,
                "position": undefined,
                "right": undefined,
                "top": undefined,
                "width": undefined,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 13,
                  },
                  undefined,
                  Object {
                    "fontFamily": "feather",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              Ôáê
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
